token/program-2022/src/extension.rs

```diff
@@ -38,30 +229,104 @@ pub enum Extension {
     /// Includes an optional mint close authority
     MintCloseAuthority,
 }
+impl TryFrom<u8> for ExtensionType {
+    type Error = ProgramError;
 
-/// Type-Length-Value Entry, used to encapsulate all extensions contained within an account
-#[repr(C)]
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct TlvEntry<V> {
-    /// Extension type encoded in the rest of the entry
-    pub extension: Extension,
-    /// Length of the entry, in bytes
-    pub length: u32,
-    /// Deserialized value
-    pub value: V,
+    fn try_from(extension_type: u8) -> Result<Self, Self::Error> {
+        match extension_type {
+            0 => Ok(ExtensionType::Uninitialized),
+            1 => Ok(ExtensionType::MintTransferFee),
+            2 => Ok(ExtensionType::AccountTransferFee),
+            3 => Ok(ExtensionType::MintCloseAuthority),
+            _ => Err(ProgramError::InvalidAccountData),
+        }
+    }
+}
+impl ExtensionType {
+    /// Get the data length of the type associated with the enum
+    pub fn get_associated_type_len(&self) -> usize {
```

Comment:
With a `Pod`, this function melts away: https://github.com/solana-labs/spl-zk-token/blob/3cc9415b9dcff3c4e380369023211189f9dfc8f3/program/src/pod.rs#L47-L49

Reply:
I may be dense -- how would that work? The idea is to get the required size for an account with a certain set of extensions, denoted by their extension type enum.  This approach creates a link from the enum to the type represented underneath.  For example, `ExtensionType::MintTransferFee` is giving the size of `struct MintTransferFee`.

Reply:
Fine, I might have been too eager in this case with my Pod shilling.